File: exercism/hamming.ml
type nucleotide = A | C | G | T

let hamming_distance stranda strandb =
  if List.length stranda = List.length strandb then
    Ok
      (List.fold_left2
         (fun cnt a b -> cnt + Bool.to_int (a <> b))
         0 stranda strandb)
  else Error "strands must be of equal length"

File: exercism/hamming.js
export const compute = (strand_a, strand_b) => {
    if (strand_a.length != strand_b.length) {
	throw new Error("strands must be of equal length");
    }
    return [...strand_a].reduce((acc, a, i) => {
        return acc + ((a === strand_b[i]) ? 0 : 1);
    }, 0);
};
